packages/as-proto-gen/src/file-context.ts

```diff
@@ -40,12 +45,21 @@ export class FileContext {
 
     importNames.set(importName, uniqueImportName);
     this.registeredImports.set(importPath, importNames);
+    this.importNames.add(uniqueImportName);
 
     return [uniqueImportName, ...importNamespace].join(".");
   }
 
-  registerDefinition(definitionNamePath: string): string {
-    const [definitionName] = definitionNamePath.split(".");
+  hasImportName(importName: string): boolean {
+    return this.importNames.has(importName);
+  }
+
+  registerDefinition(definitionNamePath: string, namespace?: string): string {
```

Comment:
I think it's because of [this line](https://github.com/piotr-oles/as-proto/blob/bf354d5b14a5e0e7a4958d555362de27075a964b/packages/as-proto-gen/src/file-context.ts#L48). If we append the namespace outside of this function, it will blend with the identifier and the `definitionName` will be incorrect.